keras_nlp/models/opt/opt_backbone.py

```diff
@@ -0,0 +1,181 @@
+        dropout: float. Dropout probability for the Transformer encoder.
+        max_sequence_length: int. The maximum sequence length that this encoder
+            can consume. If None, `max_sequence_length` uses the value from
+            sequence length. This determines the variable shape for positional
+            embeddings.
+
+    Examples:
+    ```python
+    input_data = {
+        "token_ids": tf.ones(shape=(1, 12), dtype=tf.int64),
+        "padding_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)
+        ),
+    }
+    # Randomly initialized OPT decoder model with a custom config
+    model = keras_nlp.models.OPTBackbone(
+        vocabulary_size=50265,
+        num_layers=6,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12,
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        embedding_dim,
+        hidden_dim,
+        intermediate_dim,
+        dropout=0.1,
+        max_sequence_length=1024,
+        **kwargs,
+    ):
+        # Decoder inputs.
+        token_ids = keras.Input(shape=(None,), dtype="int32", name="token_ids")
+        padding_mask = keras.Input(
+            shape=(None,), dtype="int32", name="padding_mask"
+        )
+
+        # Embed tokens, potentially at a lower dimensionality.
+        token_embedding = keras.layers.Embedding(
+            input_dim=vocabulary_size,
+            output_dim=embedding_dim,
+            embeddings_initializer=opt_kernel_initializer(),
+            name="token_embedding",
+        )(token_ids)
+        # Project tokens to the hidden_dim in necessary.
+        if embedding_dim != hidden_dim:
+            token_embedding = keras.layers.Dense(
+                hidden_dim,
+                kernel_initializer=opt_kernel_initializer(),
+                use_bias=False,
+                name="embedding_projection",
+            )(token_embedding)
+        # Embed positions as the hidden_dim.
+        position_embedding = PositionEmbedding(
+            initializer=opt_kernel_initializer(),
+            sequence_length=max_sequence_length,
+            name="position_embedding",
+        )(token_embedding)
+        # Sum embeddings.
+        x = keras.layers.Add()((token_embedding, position_embedding))
+
+        # Apply successive transformer decoder blocks.
+        for i in range(num_layers):
+            x = TransformerDecoder(
+                intermediate_dim=intermediate_dim,
+                num_heads=num_heads,
+                dropout=dropout,
+                activation=lambda x: keras.activations.gelu(
+                    x, approximate=False
+                ),
+                layer_norm_epsilon=1e-5,
+                kernel_initializer=opt_kernel_initializer(),
+                name=f"transformer_layer_{i}",
+                has_cross_attention=True,
```

Comment:
Why is this needed for a decoder-only model?

Reply:
just a copy pasta error :)